langchain-core/src/utils/types/zod.ts

```diff
@@ -781,3 +785,27 @@ export function interopZodTransformInputSchema(
 
   throw new Error("Schema must be an instance of z3.ZodType or z4.$ZodType");
 }
+
+/**
+ * Sanitizes a Zod schema by transforming it to its input schema and then making it strict.
+ * Supports both Zod v3 and v4 schemas. If `recursive` is true, applies strictness recursively to all nested object schemas and arrays of object schemas.
+ *
+ * @param schema - The Zod schema instance (v3 or v4)
+ * @param {boolean} [recursive=false] - Whether to recursively process nested objects/arrays.
+ * @returns The sanitized Zod schema.
+ */
+export function interopZodSanitizeSchema(
```

Comment:
Extracted so that it can be called independently [here](https://github.com/langchain-ai/langchainjs/pull/9125/files#diff-fddcb19e8c7c2718f4e2d6476956f4c3d5b3bc9c020ec5ca28301d04c0bab974R1246).